server/src/main/java/org/opensearch/common/concurrent/GatedAutoCloseable.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
```

Comment:
If these are new files they shouldn't have the Elastic Apache Header. Only files inherited (or renamed) from Elasticsearch should have this header. All new files should only have the SPDX OpenSearch License Header.